### DISCUSSION TITLE
`public-hoist-pattern` for workspace package

### DISCUSSION THREAD

**Discussion by vjpr** [pnpm/pnpm#3496] [Ideas]
Assume `hoist=false`.

Gatsby for example builds its app to the `<workspace-package>/.cache` dir. There are usually many deps that are missing from these SSR bundles.

I don't like to use `public-hoist-patterns` because it pollutes the `node_modules` causing unexpected things. Same goes for `hoist=false`.

But how about allowing this at the workspace package level?

So some packages could have a flat `node_modules` but not all.

Does this already exist? Thoughts?

**Comment by zkochan** [comment]
It is not possible to implement because the real location of dependencies is at `<workspace root>/node_modules/.pnpm`, so they will not look for the hoisted dependencies in project directories.

**Reply by vjpr** [reply]
To clarify, I'm talking about a project structure like below:

```
myproject
  build
    .ssr
      index.js - require('foo') but is not defined in `package.json
      (node_modules) <----- NOTE: DOESN'T EXIST
  package.json - NOTE: 'foo' is not listed as dependency here.
  node_modules
```

_myproject-public-hoist-pattern[]=foo_

```
myproject
  build
    .ssr
      index.js
  package.json
  node_modules
    foo <--- foo is installed here so that `build/.ssr/index.js` can access it
```

Probably easier to fix these issues with `.pnpmfile.cjs` though...

I think you meant that transitive deps can't access hoisted deps using this approach?

**Reply by zkochan** [reply]
for application code it would work but if it is application code, why don't you just fix the app? hoisting is for buggy dependencies.
